api/src/Entity/User.php

```diff
@@ -70,74 +71,6 @@ class User extends BaseEntity implements UserInterface, PasswordAuthenticatedUse
     #[ApiProperty(readable: false, writable: false)]
     public Collection $collaborations;
```

Comment:
`ownedCamps`, `password` und weitere Properties hängen jetzt immer noch am User, sind also nur daher nicht für die "Öffentlichkeit" exponiert weil sie readable: false sind. Ist das korrekt so, oder wären die nicht besser beim Profile aufgehoben?

Denselben Gedanken hatte ich für die `collaborations`, aber die müssen ja auf dem User sein damit man neue User in ein Lager einfügen kann. Oder wird das durch das Einladungs-System abgefangen, und auch die könnte man aufs Profile verschieben?

Reply:
Wenn ich als zukünftiger Entwickler die Applikation anschaue, würde ich weder username noch passwort auf der Profile Entity erwarten. Dies ist jetzt nur so komisch, weil wir diese Abstufung der Sichtbarkeit wollen.

Ich habe mir sogar überlegt, den username auf dem User zu lassen, und dann vom Profile her über den User zu holen, habe mich dann aber dagegen entschieden, da das noch komischer wäre.

Ownedcamps denke ich können wir bald auch abschaffen, das braucht es nicht wirklich momentan. Würde ich aber nicht in diesem PR machen.

Jetzt habe ich einfach das verschoben, was wir nur auf dem profile wollen. Alles was so oder so nicht sichbar ist (und nicht sichtbar bleiben wird), bleibt beim User..
Das verschieben wäre jetzt nicht mehr so schwierig, da kann man die migration analog schreiben.
Die Idee momenan war (leider nicht ganz durchgezogen), dass aus sicht des php codes der User noch so funktioniert wie bisher. Das macht es auch einfacher, wenn wir das Profile dann noch weiter aufteilen wollen.

Die collaborations aufs profile zu schieben, weil wir die sichtbarkeit der properties abstufen wollen, macht für mich nicht wirklich sinn. Die beziehung besteht zum user, und nicht zum profile. Das auflisten der "campCollaborations" haben wir schon, indem wir die für den eigenen user sichtbaren camps filtern. Und dass man von einem anderen User sehen möchte, in welchen camps der ist, wird so bald nicht nötig sein, denke ich.

"
Denselben Gedanken hatte ich für die collaborations, aber die müssen ja auf dem User sein damit man neue User in ein Lager einfügen kann. Oder wird das durch das Einladungs-System abgefangen, und auch die könnte man aufs Profile verschieben?
"
Könnte man, das Einladungssystem fängt das nicht ab, aber man könnte das so schreiben. Macht für mich aber nicht mehr sinn, als so wie es momentan ist.